otter/integration/tests/test_pause.py

```diff
@@ -0,0 +1,187 @@
+"""
+Tests covering `../groups/groupId/pause` and `../groups/groupId/resume`
+endpoints
+"""
+from testtools.matchers import ContainsDict, Equals
+
+import treq
+
+from twisted.internet import reactor
+from twisted.internet.defer import inlineCallbacks, returnValue
+from twisted.trial import unittest
+
+from otter.integration.lib.autoscale import ScalingPolicy
+from otter.integration.lib.mimic import MimicNova
+from otter.integration.lib.resources import TestResources
+from otter.integration.lib.trial_tools import (
+    TestHelper,
+    convergence_interval,
+    get_identity,
+    get_resource_mapping,
+    get_utcstr_from_now,
+    not_mimic,
+    region,
+    scheduler_interval,
+    skip_if,
+    sleep
+)
+
+
+timeout_default = 600
+
+
+class PauseTests(unittest.TestCase):
+    """
+    Tests for `../groups/groupId/pause` endpoint
+    """
+
+    skip = "Until #1604 is implemented"
+
+    def setUp(self):
+        self.helper = TestHelper(self)
+        self.rcs = TestResources()
+        self.identity = get_identity(self.helper.pool)
+        return self.identity.authenticate_user(
+            self.rcs,
+            resources=get_resource_mapping(),
+            region=region
+        )
+
+    @inlineCallbacks
+    def test_pause_getstate(self):
+        """
+        A paused group's state will say paused:True
+        """
+        group, _ = self.helper.create_group()
+        yield group.start(self.rcs, self)
+        yield self.helper.assert_group_state(
+            group, ContainsDict({"paused": Equals(False)}))
+        yield group.pause(self.rcs)
+        yield self.helper.assert_group_state(
+            group, ContainsDict({"paused": Equals(True)}))
+
+    @skip_if(not_mimic, "This requires mimic for long builds")
+    @inlineCallbacks
+    def test_pause_stops_convergence(self):
+        """
+        Pausing a group will stop any further convergence cycle. We do this by
+        1. Setup mimic to build server for just before timeout
+        2. Creating a group with 1 min entity
+        3. Pause the group
+        4. Wait for server to finish building in mimic
+        5. Notice that group continues to think that server is building
+        """
+        mimic_nova = MimicNova(pool=self.helper.pool, test_case=self)
+        server_build_time = convergence_interval + 5
+        yield mimic_nova.sequenced_behaviors(
+            self.rcs,
+            criteria=[{"server_name": "pause-stops-convergence" + ".*"}],
+            behaviors=[
+                {"name": "build",
+                 "parameters": {"duration": server_build_time}}
+            ])
+        group, _ = self.helper.create_group(min_entities=1)
+        yield group.start(self.rcs, self)
+        one_building = ContainsDict({"pendingCapacity": Equals(1),
+                                     "activeCapacity": Equals(0),
+                                     "status": Equals("ACTIVE")})
```

Comment:
Similarly to https://github.com/rackerlabs/otter/pull/1642/files#r36238678, would it make sense to also add a `desiredCapacity` check here?